### PR TITLE
[TE] Yyuen/add ember

### PR DESCRIPTION
**Pull request by justYves**
### What's new:
- imported Ember-CLI and Ember
- implemented pods structure
- integrated ember build to the maven install process
- allow back end for CORS request in dev environment
- added ember-redux with a simple example page 

### To do (in follow up pr): 
- add ember-mirage 
- working basic integration test 
- include ember test as part of the maven build

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/common/ThirdEyeConfiguration.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
 
   private String phantomJsPath = "";
 
+  private boolean cors = false;
```
Comment (justYves) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/common/ThirdEyeConfiguration.java]: Thank you @npawar  for your help on this

File: thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/component.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import Ember from 'ember';
```
Comment (yee-lkd) [inline thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/component.js]: Can we please rename this component to "anomaly-graph.js" `../thirdeye-frontend/app/pods/components/anomaly-graph.js`? Conventionally, Ember creates 3 files when one creates a component through Ember CLI. `.../components/component-name.js`, `.../templates/components/component-name.hbs`, and `../tests/integration/components/component-name.js`. Would be good to be consistent with Ember naming/structure conventions :)

File: thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/template.hbs [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<li><strong>Name:</strong> {{anomaly.anomalyFunctionName}}</li>
```
Comment (yee-lkd) [inline thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/template.hbs]: Please see comment regarding component name above.

File: thirdeye/thirdeye-frontend/app/pods/components/containers/anomaly-container/component.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+import Ember from 'ember';
```
Comment (yee-lkd) [inline thirdeye/thirdeye-frontend/app/pods/components/containers/anomaly-container/component.js]: Please see comment regarding component name above.

File: thirdeye/thirdeye-frontend/app/router.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import Ember from 'ember';
+import config from './config/environment';
+
+const Router = Ember.Router.extend({
+  location: config.locationType,
+  rootURL: config.rootURL
+});
+
+Router.map(function() {
+  this.route('example', { path: 'example/:id'});
```
Comment (yee-lkd) [inline thirdeye/thirdeye-frontend/app/router.js]: Nitpick - space after `id'`

File: thirdeye/thirdeye-frontend/tests/helpers/resolver.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import Resolver from '../../resolver';
+import config from '../../config/environment';
```
Comment (yee-lkd) [inline thirdeye/thirdeye-frontend/tests/helpers/resolver.js]: It's usually better to use absolute path to be sure that we import from the right place. Also, it's more readable (or clear) where we're importing from.

File: thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/component.js (hunk 2 above)
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/component.js]: @yee-lkd those files were actually generated with `ember g`. This is the pod structure default's naming convention i.e. `components/component-name/template.hbs` `components/component-name/component.js`

File: thirdeye/thirdeye-frontend/app/router.js (hunk 5 above)
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/router.js]: good eye!

File: thirdeye/thirdeye-frontend/tests/helpers/resolver.js (hunk 6 above)
Comment (justYves) [inline thirdeye/thirdeye-frontend/tests/helpers/resolver.js]: Looks like this is auto generated by ember. The same way  
```
import startApp from '../helpers/start-app';
import destroyApp from '../helpers/destroy-app';
```
is generated for acceptance test. Is this a setting I can modify to change those to absolute path?

File: thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/template.hbs (hunk 3 above)
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/anomaly-graph/template.hbs]: see pod structure comment above

File: thirdeye/thirdeye-frontend/app/pods/components/containers/anomaly-container/component.js (hunk 4 above)
Comment (justYves) [inline thirdeye/thirdeye-frontend/app/pods/components/containers/anomaly-container/component.js]: see pod structure comment above

File: thirdeye/thirdeye-frontend/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+* `ember test`
+* `ember test --server`
+
+### Building
+
+* `ember build` (development)
+* `ember build --environment production` (production)
+
+### Deploying
```
Comment (kishoreg) [inline thirdeye/thirdeye-frontend/README.md]: can you add instructions on how a dev can bring up this server locally?

Comment (justYves) [inline thirdeye/thirdeye-frontend/README.md]: line 23 lists the instructions: 
```
  +## Running / Development
  +* `ember serve`
  +* Visit your app at [http://localhost:4200](http://localhost:4200)
```

I'll also update our internal confluence page

**Pull request merged**